streams/src/test/java/org/apache/kafka/streams/StreamsConfigTest.java

```diff
@@ -31,13 +33,15 @@
 
     private Properties props = new Properties();
     private StreamsConfig streamsConfig;
-    private StreamThread streamThreadPlaceHolder;
-
 
     @Before
     public void setUp() {
         props.put(StreamsConfig.APPLICATION_ID_CONFIG, "streams-config-test");
         props.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
+        props.put(StreamsConfig.KEY_SERDE_CLASS_CONFIG, Serdes.String().getClass().getName());
+        props.put(StreamsConfig.VALUE_SERDE_CLASS_CONFIG, Serdes.String().getClass().getName());
+        props.put("key.deserializer.encoding", "UTF8");
+        props.put("value.deserializer.encoding", "UTF-16");
```

Comment:
It will be a bit hard to validate directly that these properties are used. So in the added unit test, we use UTF-16 for value serde, and if the properties are not used then the deser-after-ser will not pass.